src/views/course_pairings.vue

```diff
@@ -0,0 +1,239 @@
+            user_associations: [],
+            list_courses: [],
+            list_terms: [],
+            tableSettings: {
+                data: [],
+                stretchH: "all",
+                colHeaders: []
+            }
+        };
+    },
+    /**
+     * Function is called when page is created to get course pairings from API.
+     * Data is used to populate the spreadsheet and options for select
+     */
+    created: function() {
+        this.getData();
+    },
+
+    watch: {
+        /**
+         * Function is called when specify_course is modified. The data is used
+         * to populate the page with the specified course.
+         */
+        specify_course: function() {
+            this.getData();
+        },
+
+        /**
+         * Function is called when specify_term is modified. The data is used
+         * to populate the page with the specified term.
+         */
+        specify_term: function() {
+            this.getData();
+        },
+
+        /**
+         * Function is called when column is modified. The data is used
+         * to populate the page with the specified column.
+         */
+        column: function() {
+            this.getData();
+        }
+    },
+    methods: {
+        /**
+         * Function gets course pairing data from API for the specified
+         * course, term and column and calls parseData to parse retuned data.
+         */
+        getData: function() {
+            // Converting to parameter enums
+            var column_name = this.column === "TA" ? "ta" : "instructor";
+
+            var course_where = "";
+            var term_where = "";
+            if (this.specify_term != "All Terms") {
+                term_where = `${this.specify_term}`;
+            }
+            if (this.specify_course != "All Courses") {
+                course_where = `${this.specify_course}`;
+            }
+
+            let url =
+                "{" +
+                '"what":"course_pairings",' +
+                `"user_id":"${this.$route.params.user_id}",` +
+                `"term":"${term_where}",` +
+                `"course_code":"${course_where}",` +
+                `"column":"${column_name}"` +
+                "}";
+
+            url = JSON.parse(url);
+            fetch(generate_get_url(url))
+                .then(res => res.json())
+                .then(data => this.parseData(data))
+                .catch(err => {
+                    this.$emit("error", err.toString());
+                });
+        },
+
+        /**
+         * Function parses data from API call that will be rendered on the
+         * spreadsheet and select component.
+         */
+        parseData: function(data) {
+            var pairings = {};
+            for (let element of data.DATA) {
+                // Create an associative array with keys as course code and values
+                // as an array of instructors/tas for that couse.
+                if (element.course_code in pairings) {
```

Comment:
There's a short way to do this in JS.

    pairings[element.course_code]  = pairings[element.course_code] || []
    pairings[element.course_code].push(...)